File: Data Structures & Algorithms/validate-parentheses/submission-6.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        pair = {'(': ')', '[': ']', '{' : '}'}
        for c in s:
            if c in (')', '}', ']'):
                if len(stack) == 0:
                    return False
                if stack[-1] != c:
                    return False
                stack.pop()
                continue

            stack.append(pair[c])
        
        return len(stack) == 0

File: Data Structures & Algorithms/minimum-stack/submission-6.cpp
class MinStack {
public:
    std::stack<int> main;
    std::stack<int> min;

    MinStack() {
        
    }
    
    void push(int val) {
        main.push(val);
        if (min.empty() || val <= min.top())
            min.push(val);
    }
    
    void pop() {
        if (main.top() == min.top())
            min.pop();
        main.pop();
    }
    
    int top() {
        return main.top();
    }
    
    int getMin() {
        return min.top();
    }
};
